apps/full-site-editing/webpack.config.js

```diff
@@ -57,6 +71,24 @@ function getWebpackConfig( env = {}, argv = {} ) {
 			path: outputPath,
 			filename: '[name].js', // dynamic filename
 		},
+		plugins: [
+			...webpackConfig.plugins.filter(
+				( plugin ) => plugin.constructor.name !== 'DependencyExtractionWebpackPlugin'
+			),
+			new DependencyExtractionWebpackPlugin( {
+				injectPolyfill: true,
+				requestToExternal( request ) {
+					if ( request.startsWith( FSE_MODULE_PREFIX ) ) {
+						switch ( request ) {
+							case 'a8c-fse-common-data-stores':
```

Comment:
Could have documentation pointing out where this name comes from since there's no convention of "handle matches a package in NPM"?

Reply:
This is the roughest spot here and I'm not sure what the best approach is. Documentation, at least, is a good start 👍

Reply:
Would it be possible to create an actual module that relies on the globally registered script?

A bit less magic-y.

Reply:
Would it be possible to create an actual module that relies on the globally registered script?

A bit less magic-y.

Reply:
> Would it be possible to create an actual module that relies on the globally registered script?
> A bit less magic-y.

I'm generally in favor of reducing magic, but I'm not sure of a good way to do that here.

This quickly gets into some tricky webpack territory…

What would the import be? We have an actual module that we want to use, but we don't have a good way of referring to it. What are modules? Names registered on npm? Mapped aliases in webpack? Paths to files on local machines?

Ostensibly, we'd like to import the module that registers the data stores. It's in the repository and we can import it via relative imports, but not likely in a consistent way. With the current [`dependency-extraction-webpack-plugin` this may be impossible, I suspect if we were to change the `requestToExternal` API to provide the context](https://github.com/WordPress/gutenberg/blob/b2917b4906699a2085364fde4c6b9f6195618de4/packages/dependency-extraction-webpack-plugin/index.js#L45-L68), we might be able to determine that we're importing a specific file.

I believe this is all complicated by the fact that FSE JavaScript is output by a bunch of independent, unconnected webpack builds, so we need to stitch things together from outside. If we were to build everything together from a number of entrypoints, we might gain finer control over this sort of thing. But I'm speculating 🙂 

In short, this could be better and we can probable polish things or improve the conventions. I don't know a great way of doing that.

I _did_ add a faux module for TypeScript, so the typechecker will complain unknown modules are imported, and it understands that `a8c-fse-common-data-stores` is a module with no exports.